content/HoarderCard.tsx

```diff
@@ -23,39 +27,80 @@ export function HoarderCard({ className, userQuery }: { className?: string; user
 
   if (!options.apiKey || !options.url) {
     return (
-      <div className="rounded-md bg-background p-2">
-        <h2 className="text-center font-bold text-foreground text-lg">
-          Hoarder's Pipette: Please open options page to configure your API key and URL
-        </h2>
-      </div>
+      <Card className="w-full max-w-2xl">
+        <CardHeader className="border-gray-200 border-b dark:border-gray-700">
+          <h2 className="font-semibold text-foreground text-lg">Hoarder's Pipette</h2>
+        </CardHeader>
+        <CardContent className="p-6 text-center">
+          <p className="text-foreground/80">Please open options page to configure your API key and URL</p>
+        </CardContent>
+      </Card>
     )
   }
 
   if (!userQuery) {
     return null
   }
 
-  if (!bookmarks || bookmarks.length === 0) {
-    return null
-  }
+  const hasBookmarks = bookmarks && bookmarks.length > 0
 
   return (
-    <Card className={className}>
-      <CardHeader>
-        <h2 className="font-bold text-xl">Karakeep Bookmarks</h2>
+    <Card className={cn(className, 'w-full max-w-2xl')}>
+      <CardHeader className="border-gray-200 border-b px-6 py-4 dark:border-gray-700">
+        <div className="flex items-center justify-between">
+          <h2 className="font-semibold text-foreground text-xl">Karakeep Bookmarks</h2>
+        </div>
       </CardHeader>
-      <CardContent>
-        <ScrollArea className="@container h-72 pr-8">
-          <div className="flex flex-col gap-2">
-            {pipe(
-              bookmarks, // Use the bookmarks array
-              Array.filter((bookmark) => bookmark.content.type === 'link'),
-              Array.map((bookmark) => <BookmarkPreview key={bookmark.id} bookmark={bookmark} />),
-            )}
-          </div>
-          <ScrollBar orientation="vertical" />
-        </ScrollArea>
+
+      <CardContent className="p-0">
+        {(() => {
```

Comment:
💭  I prefer to avoid using IIFEs. Embedding logic directly like this makes me worry about accidentally misunderstanding the scope of things like `return` or `if` statements. For example, where exactly does this `return` return to? Do you think we could break this down into separate components?